pkg/api/types.go

```diff
@@ -1414,8 +1412,13 @@ type Toleration struct {
 	// Effect indicates the taint effect to match. Empty means match all taint effects.
 	// When specified, allowed values are NoSchedule and PreferNoSchedule.
 	Effect TaintEffect `json:"effect,omitempty"`
-	// TODO: For forgiveness (#1574), we'd eventually add at least a grace period
-	// here, and possibly an occurrence threshold and period.
+	// ForgivenessSeconds Represents the period of time the toleration tolerates the taint.
```

Comment:
Change the comment on line 1413 to say "When specified, allowed values are NoSchedule, PreferNoSchedule, and NoExecute". Similar change on line 1371 (add NoExecute).

The comment here (starting on line 1416) is correct but I think it could be phrased a little more clearly/concisely. How about
```
ForgivenessSeconds represents the period of time the toleration (which must be of effect NoExecute,
otherwise this field is ignored) tolerates the taint. By default, it is not set, which means tolerate the taint
forever (do not evict). Zero and negative values are not allowed.
```